app/models/api/template.js

```diff
@@ -15,12 +15,25 @@ function TemplateViewModel(data) {
   this.RestartPolicy = data.restart_policy ? data.restart_policy : 'always';
   this.Volumes = [];
   if (data.volumes) {
+    var self = this;
     this.Volumes = data.volumes.map(function (v) {
-      return {
+      var volume = {
         readOnly: false,
-        containerPath: v,
+        containerPath: v.container,
         type: 'auto'
       };
+
+      if (v.bind) {
+        volume.name = v.bind;
+        volume.type = 'bind';
+      }
+
+      if (typeof v === 'string') {
+        volume.containerPath = v;
+        self.DeprecatedFormat = true;
```

Comment:
The use of `self` here is pretty limited as it's only declared and use for the declaration of the `DeprecatedFormat` variable. I'd rather see `this.DeprecatedFormat = true;` here.